doc/devel/testing.rst

```diff
@@ -4,31 +4,34 @@
 Developer's tips for testing
 ============================
 
-Matplotlib has a testing infrastructure based on nose_, making it easy
-to write new tests. The tests are in :mod:`matplotlib.tests`, and
-customizations to the nose testing infrastructure are in
-:mod:`matplotlib.testing`. (There is other old testing cruft around,
-please ignore it while we consolidate our testing to these locations.)
-
-.. _nose: https://nose.readthedocs.io/en/latest/
+Matplotlib has a testing infrastructure based on pytest_, making it easy to
+write new tests. The tests are in :mod:`matplotlib.tests`, and customizations
+to the pytest testing infrastructure are in :mod:`matplotlib.testing`. (There
+is other old testing cruft around, please ignore it while we consolidate our
+testing to these locations.)
+
+.. _pytest: http://doc.pytest.org/en/latest/
+.. _mock: https://docs.python.org/dev/library/unittest.mock.html>
+.. _Ghostscript: https://www.ghostscript.com/
+.. _Inkscape: https://inkscape.org
+.. _pytest-cov: https://pytest-cov.readthedocs.io/en/latest/
+.. _pytest-pep8: https://pypi.python.org/pypi/pytest-pep8
 
 Requirements
 ------------
 
 The following software is required to run the tests:
 
-  - nose_, version 1.0 or later
-  - `mock <https://docs.python.org/dev/library/unittest.mock.html>`_, when running python
-    versions < 3.3
-  - `Ghostscript <https://www.ghostscript.com/>`_ (to render PDF
-    files)
-  - `Inkscape <https://inkscape.org>`_ (to render SVG files)
+  - pytest_, version 3.0.0 or later
+  - mock_, when running Python versions < 3.3
+  - Ghostscript_ (to render PDF files)
+  - Inkscape_ (to render SVG files)
 
 Optionally you can install:
 
-  - `coverage <https://coverage.readthedocs.io/en/latest/>`_ to collect coverage
-    information
-  - `pep8 <http://pep8.readthedocs.io/en/latest>`_ to test coding standards
+  - pytest-cov_ to collect coverage information
+  - pytest-pep8_ to test coding standards
```

Comment:
I would add all the optional dependencies here:
   -  pytest-xdist
   -  pytest-timeout_